app-center-services/src/main/java/org/exoplatform/appcenter/search/ApplicationESListener.java

```diff
@@ -0,0 +1,35 @@
+package org.exoplatform.appcenter.search;
+
+import org.exoplatform.appcenter.util.Utils;
+import org.exoplatform.commons.search.index.IndexingService;
+import org.exoplatform.commons.utils.CommonsUtils;
+import org.exoplatform.services.listener.Event;
+import org.exoplatform.services.listener.Listener;
+import org.exoplatform.services.log.ExoLogger;
+import org.exoplatform.services.log.Log;
+
+public class ApplicationESListener extends Listener<Long, Object> {
+    private static final Log LOG = ExoLogger.getExoLogger(ApplicationESListener.class);
+
+    private final IndexingService indexingService;
+
+    public ApplicationESListener(IndexingService indexingService) {
+        this.indexingService = indexingService;
+    }
+
+    @Override
+    public void onEvent(Event<Long, Object> event) throws Exception {
+        if (indexingService != null) {
+            String applicationId = Long.toString(event.getSource());
+            if (Utils.POST_CREATE_APPLICATION_EVENT.equals(event.getEventName())) {
```

Comment:
What about updating/deleting applications ? shouldn't we taka care of those events ?

Reply:
i'll take care of those